File: Core/Models/Games/TicTacToeGameState.cs
using Core.Models.Interfaces;

namespace Core.Models.Games
{
    public class TicTacToeGameState : IGameState
    {
        public bool HasGameBeenSetup { get; set; }
        public string[] Board { get; set; }
        public bool IsPlayer1Turn { get; set; }
        public bool GameIsPlayable { get; set; }
        public string Result { get; set; }

        public TicTacToePlayer Player1 { get; set; }
        public TicTacToePlayer Player2 { get; set; }
    }
}

File: UI.Web/ClientApp/src/app/models/TicTacToeGameState.ts
import { TicTacToePlayer } from "./TicTacToePlayer";

export class TicTacToeGameState{
    constructor(){    }
    
    hasGameBeenSetup: boolean;
    board:string[];
    gameIsPlayable: boolean;
    isPlayer1Turn: boolean;
    player1: TicTacToePlayer;
    player2: TicTacToePlayer;
}